engines/sherlock/sherlock.cpp

```diff
@@ -0,0 +1,295 @@
+	_journal = new Journal(this);
+	_people = new People(this);
+	_saves = new SaveManager(this, _targetName);
+	_scene = new Scene(this);
+	_screen = new Screen(this);
+	_sound = new Sound(this, _mixer);
+	_talk = new Talk(this);
+	_ui = new UserInterface(this);
+
+	// Load game settings
+	loadConfig();
+}
+
+/**
+ * Main method for running the game
+ */
+Common::Error SherlockEngine::run() {
+	// Initialize the engine
+	initialize();
+
+	// Flag for whether to show original saves dialog rather than the ScummVM GMM
+	_showOriginalSavesDialog = ConfMan.hasKey("OriginalSaves") && ConfMan.getBool("OriginalSaves");
+
+	// If requested, load a savegame instead of showing the intro
+	if (ConfMan.hasKey("save_slot")) {
+		int saveSlot = ConfMan.getInt("save_slot");
+		if (saveSlot >= 1 && saveSlot <= MAX_SAVEGAME_SLOTS)
+			_loadGameSlot = saveSlot;
+	}
+
+	if (_loadGameSlot != -1) {
+		_saves->loadGame(_loadGameSlot);
+		_loadGameSlot = -1;
+	} else {
+		showOpening();
+	}
+
+	while (!shouldQuit()) {
+		// Prepare for scene, and handle any game-specific scenes. This allows
+		// for game specific cutscenes or mini-games that aren't standard scenes
+		startScene();
+		if (shouldQuit())
+			break;
+
+		// Reset UI flags
+		_ui->reset();
+
+		// Reset the data for the player character (Sherlock)
+		_people->reset();
+
+		// Initialize and load the scene.
+		_scene->selectScene();
+
+		// Scene handling loop
+		sceneLoop();
+	}
+
+	return Common::kNoError;
+}
+
+/**
+ * Main loop for displaying a scene and handling all that occurs within it
+ */
+void SherlockEngine::sceneLoop() {
+	while (!shouldQuit() && _scene->_goToScene == -1) {
+		// See if a script needs to be completed from either a goto room code,
+		// or a script that was interrupted by another script
+		if (_talk->_scriptMoreFlag == 1 || _talk->_scriptMoreFlag == 3)
+			_talk->talkTo(_talk->_scriptName);
+		else
+			_talk->_scriptMoreFlag = 0;
+
+		// Handle any input from the keyboard or mouse
+		handleInput();
+
+		if (_people->_hSavedPos.x == -1) {
+			_canLoadSave = true;
+			_scene->doBgAnim();
+			_canLoadSave = false;
+		}
+	}
+
+	_scene->freeScene();
+	_people->freeWalk();
+
+}
+
+/**
+ * Handle all player input
+ */
+void SherlockEngine::handleInput() {
+	_canLoadSave = true;
+	_events->pollEventsAndWait();
+	_canLoadSave = false;
+
+	// See if a key or mouse button is pressed
+	_events->setButtonState();
+
+	_ui->handleInput();
+}
+
+/**
+ * Read the state of a global flag
+ */
+bool SherlockEngine::readFlags(int flagNum) {
+	bool value = _flags[ABS(flagNum)];
+	if (flagNum < 0)
+		value = !value;
+
+	return value;
+}
+
+/**
+ * Sets a global flag to either true or false depending on whether the specified
+ * flag is positive or negative
+ */
+void SherlockEngine::setFlags(int flagNum) {
+	_flags[ABS(flagNum)] = flagNum >= 0;
+
+	_scene->checkSceneFlags(true);
+}
+
+/**
+ * Load game configuration esttings
+ */
+void SherlockEngine::loadConfig() {
+	// Load sound settings
```

Comment:
Done. As for fade style, good catch. I actually changed the development during early development, since I personally didn't like waiting the roughly 2 seconds for a scene transition to occur. I'll change it over once I figure where to do the screen clear before the transition starts.